Improve params documentation (#5800)
nextflow-io/nextflow: docs/cli.md

### Pipeline parameters

> Pipeline scripts can define *parameters* which can be overridden on the command line.

Reviewer: ```suggestion
Pipeline scripts can define *parameters* that can be overridden on the command line.
```

Parameters can be declared in the main script:

```nextflow
params.alpha = 'default script value'

workflow {
  println "alpha = ${params.alpha}"
}
```

Or in a config file:

```groovy
params {
  alpha = 'default config value'
}
```

The above parameter can be specified on the command line as `--alpha`:

```console

> $ nextflow run <pipeline> --alpha Hello

Reviewer: ```suggestion
nextflow run main.nf --alpha Hello
```

Author: The `$` is just for consistency with the rest of the page. I wouldn't be opposed to reviewing them in a separate PR. They are useful for showing commands and output together, but aside from that they just get in the way of copying

Reviewer: Ahh! Apologies! Agree about a separate PR.

```

:::{note}
Parameters that are specified on the command line without a value are set to `true`.
:::

:::{note}

> Parameters that are specified on the command line with `kebab-case` names are automatically converted to `camelCase`. For exapmle, a parameter defined as `fooBar` in the pipeline script can be given on the command line as `--fooBar` or `--foo-bar`.

Reviewer: ```suggestion
Parameters specified on the command line in kebab case (e.g., `kebab-case`) are automatically converted to camel case (e.g., `camelCase`). Because of this, a parameter defined as `fooBar` in the pipeline script can be specified on the command line as `--fooBar` or `--foo-bar`.
```

Reviewer: Note that the vice versa was dropped here, so at the moment it reads that only kebab case are converted to camel case.
